framework/DataObjects/DataSet.py

```diff
@@ -0,0 +1,1853 @@
+      @ Out, rlz, dict, realization as {var:value} OR None if not found
+    """
+    assert(self._collector is not None)
+    # TODO KD Tree for faster values -> still want in collector?
+    # TODO slow double loop
+    lookingFor = match.values()
+    for r,row in enumerate(self._collector[:,tuple(self._orderedVars.index(var) for var in match.keys())]):
+      match = True
+      for e,element in enumerate(row):
+        if mathUtils.isAFloatOrInt(element):
+          match *= mathUtils.compareFloats(lookingFor[e],element,tol=tol)
+          if not match:
+            break
+        else:
+          match *= lookingFor[e] == element
+      if match:
+        break
+    if match:
+      return r,self._getRealizationFromCollectorByIndex(r)
+    else:
+      return len(self),None
+
+  def _getRealizationFromDataByIndex(self,index, unpackXArray=False):
+    """
+      Obtains a realization from the data storage using the provided index.
+      @ In, index, int, index to return
+      @ In, unpackXArray, bool, optional, True if the coordinates of the xarray variables must be exposed in the dict (e.g. if P(t) => {P:ndarray, t:ndarray})
+      @ Out, rlz, dict, realization as {var:value} where value is a DataArray with only coordinate dimensions
+    """
+    assert(self._data is not None)
+    #assert(index < len(self._data[self.sampleTag]))
+    rlz = self._data[{self.sampleTag:index}].drop(self.sampleTag).data_vars
+    rlz = self._convertFinalizedDataRealizationToDict(rlz, unpackXArray)
+    return rlz
+
+  def _getRealizationFromDataByValue(self,match, tol=1e-15, unpackXArray=False):
+    """
+      Obtains a realization from the data storage using the provided index.
+      @ In, match, dict, elements to match
+      @ In, tol, float, optional, tolerance to which match should be made
+      @ In, unpackXArray, bool, optional, True if the coordinates of the xarray variables must be exposed in the dict (e.g. if P(t) => {P:ndarray, t:ndarray})
+      @ Out, r, int, index where match was found OR size of data if not found
+      @ Out, rlz, dict, realization as {var:value} OR None if not found
+    """
+    assert(self._data is not None)
+    # TODO this could be slow, should do KD tree instead
+    mask = 1.0
+    for var,val in match.items():
+      # float instances are relative, others are absolute
+      if mathUtils.isAFloatOrInt(val):
+        # scale if we know how
+        try:
+          loc,scale = self._scaleFactors[var]
+        #except TypeError:
+        #  # self._scaleFactors is None, so set them
+        #  self._setScalingFactors(var)
+        except KeyError: # IndexError?
+        # variable doesn't have a scale factor (yet? Why not?)
+          loc = 0.0
+          scale = 1.0
+        scaleVal = (val-loc)/scale
+        # create mask of where the dataarray matches the desired value
+        mask *= abs((self._data[var]-loc)/scale - scaleVal) < tol
+      else:
+        mask *= self._data[var] == val
```

Comment:
fixed

Reply:
This fix failed because `TypeError: ufunc 'bitwise_and' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`

Reverting.

Reply:
What is the commit id?  Also, could you cast it with bool(stuff) somewhere?